lib/MusicBrainz/Server/Data/Search.pm

```diff
@@ -732,26 +732,8 @@ sub external_search
     $query = uri_escape_utf8($query);
     $type =~ s/release_group/release-group/;
 
-    my $search_url_string;
-    if (DBDefs->SEARCH_ENGINE eq 'LUCENE') {
-        my $dismax = $adv ? 'false' : 'true';
-        $search_url_string = "http://%s/ws/2/%s/?query=%s&offset=%s&max=%s&fmt=jsonnew&dismax=$dismax&web=1";
-    } else {
-        my $endpoint = "advanced";
-        if (!$adv)
-        {
-            # Solr has a bug where the dismax end point behaves differently
-            # from edismax (advanced) when the query size is 1. This is a fix
-            # for that. See https://issues.apache.org/jira/browse/SOLR-12409
-            if (split(/[\P{Word}_]+/, $query, 2) == 1) {
```

Comment:
Zas is looking into moving this logic into lua, but we need to switch to a deb package or something for openresty to get PCRE/Unicode support.